File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int L = 0, R = nums.length - 1 ;
        int mid;

        while(L <= R){
            mid = (L + R) / 2;
            if(nums[mid] > target){
                R = mid - 1;
            }
            else if(nums[mid] < target){
                L = mid + 1;
            }
            else{
                return mid;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1

        while left <= right:
            mid = left + (right - left) // 2

            if nums[mid] == target:
                return mid
            
            if nums[left] <= nums[mid]:
                #target in left half and not equal to mid
                if nums[left] <= target < nums[mid]:
                    right = mid - 1
                #not in left half then target must be in right half
                else:
                    left = mid + 1 
            #right half is sorted
            else:
                if nums[mid] < target <= nums[right]:
                    left = mid + 1 
                else:
                    right = mid -1 
        
        return -1
